### PR TITLE
Added Doxyfile.

### PR DESCRIPTION
**Pull request by gaaclarke**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -21,3 +21,5 @@ Thumbs.db
 .idea
 pubspec.lock
 .vscode/
+docs/doxygen/
+xcuserdata
```
Comment (chinmaygarde) [inline .gitignore]: Newline at end of file please.

File: docs/flutter-doxygen.css [hunk 2]
```diff
@@ -0,0 +1,7 @@
+div.contents {
+    margin-bottom: 10px;
+    padding: 12px;
+    margin-left: 12px;
+    margin-right: auto;
+    max-width: 960px;
+}
```
Comment (chinmaygarde) [inline docs/flutter-doxygen.css]: Newline at end of file please.

File: Doxyfile [hunk 3]
```diff
@@ -0,0 +1,2537 @@
...
+# the logo to the output directory.
+
+PROJECT_LOGO           =
+
+# The OUTPUT_DIRECTORY tag is used to specify the (relative or absolute) path
+# into which the generated documentation will be written. If a relative path is
+# entered, it will be relative to the location where doxygen was started. If
+# left blank the current directory will be used.
+
+OUTPUT_DIRECTORY       = docs/doxygen
```
Comment (chinmaygarde) [inline Doxyfile]: Please put this in the `../out/docs/doxygen` directory instead.

File: .gitignore [hunk 4]
```diff
@@ -21,3 +21,5 @@ Thumbs.db
 .idea
 pubspec.lock
 .vscode/
+docs/doxygen/
```
Comment (chinmaygarde) [inline .gitignore]: I'd prefer the docs went inside the `out` directory instead. That is already in the ignores and is usually the place we put generated build artifacts.

**Comment by chinmaygarde** [comment]
I believe the license script can be patched to ignore the `Doxyfile`.

File: .gitignore (hunk 1 above)
Comment (gaaclarke) [reply .gitignore]: Done.

File: docs/flutter-doxygen.css (hunk 2 above)
Comment (gaaclarke) [reply docs/flutter-doxygen.css]: Done, Visual Studio Code removes ending newlines in css files.  I'm guessing for filesize.

File: .gitignore (hunk 4 above)
Comment (gaaclarke) [reply .gitignore]: Doxygen isn't smart enough to create directories so I could only put them in ../out/html, not ../out/<some-directory>/html.  That doesn't seem ideal.

Comment (chinmaygarde) [reply .gitignore]: Sounds good.

**Pull request merged**